sos/report/plugins/proxmox.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2024 ORNESS/Ditrit Drien Breton <drien.breton@orness.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import json
+from pathlib import Path
+
+from sos.report.plugins import Plugin, DebianPlugin, PluginOpt
+
+
+class Proxmox(Plugin, DebianPlugin):
+    """
+    This plugin will capture information about the system's
+    Proxmox Virtualization Environment.
+
+    It will collect information about the cluster, nodes, pools and
+    storage from the Proxmox API.
+    """
+    short_desc = 'Proxmox cluster information'
+
+    plugin_name = 'proxmox'
+    packages = ('proxmox-ve',)
+
+    option_list = [
+        PluginOpt('output_formats',
+                  desc='List of output formats to use '
+                       'for the commands separated by ":".',
+                  default='text',
+                  ),
+        PluginOpt('cmd_override_file',
+                  desc='File containing the commands '
+                       'to run instead of the default ones.',
+                  default=None,
+                  val_type=[None, Path, str],
+                  ),
```

Comment:
Hard nack.

We have the `sos_extras` plugin to handle extra collections that aren't in plugins, and it serves entirely as a stop-gap measure before those commands get contributed and released.